Sources/NIOSSH/SshAgent/Protocol.swift

```diff
@@ -0,0 +1,168 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// SSH Agent protocol as described here:
+// https://datatracker.ietf.org/doc/html/draft-miller-ssh-agent-17
+// and using the wire representation described here:
+// https://datatracker.ietf.org/doc/html/rfc4251#section-5
+//
+
+import Foundation
+import NIOCore
+
+enum MessageNumber: UInt8, Sendable {
```

Comment:
I typically recommend making this a `struct` with a bunch of `static let`s, not an `enum`.